src/test/java/uk/gov/companieshouse/api/accounts/controller/LoansControllerTest.java

```diff
@@ -0,0 +1,335 @@
+
+        ResponseEntity responseEntity = ResponseEntity.status(HttpStatus.FOUND).body(responseObject.getDataForMultipleResources());
+        when(apiResponseMapper.mapGetResponseForMultipleResources(responseObject.getDataForMultipleResources(), request))
+                .thenReturn(responseEntity);
+
+        ResponseEntity response = controller.getAll(COMPANY_ACCOUNT_ID, request);
+
+        assertNotNull(response);
+        assertEquals(HttpStatus.FOUND, response.getStatusCode());
+        assertEquals(loans, response.getBody());
+    }
+
+    @Test
+    @DisplayName("Tests the retrieval of all Loans when the service throws a DataException")
+    void getAllLoansServiceThrowsDataException() throws DataException {
+
+        when(request.getAttribute(AttributeName.TRANSACTION.getValue())).thenReturn(transaction);
+
+        when(loanService.findAll(transaction, COMPANY_ACCOUNT_ID, request))
+                .thenThrow(DataException.class);
+
+        ResponseEntity responseEntity = ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+        when(apiResponseMapper.getErrorResponse()).thenReturn(responseEntity);
+
+        ResponseEntity response = controller.getAll(COMPANY_ACCOUNT_ID, request);
+
+        assertNotNull(response);
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, response.getStatusCode());
+        assertNull(response.getBody());
+    }
+
+    @Test
+    @DisplayName("Tests the successful update of a Loan resource")
+    void updateLoanSuccess() throws DataException {
+
+        when(request.getAttribute(anyString())).thenReturn(loansToDirectors).thenReturn(transaction);
+
+        when(loansToDirectors.getLoans()).thenReturn(loans);
+        when(loans.get(LOANS_ID)).thenReturn(LOANS_LINK);
+
+
+        ResponseObject responseObject = new ResponseObject(ResponseStatus.UPDATED, loanRest);
+        when(loanService.update(loanRest, transaction, COMPANY_ACCOUNT_ID, request))
+                .thenReturn(responseObject);
+
+        ResponseEntity responseEntity = ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+        when(apiResponseMapper.map(responseObject.getStatus(), responseObject.getData(), responseObject.getErrors()))
+                .thenReturn(responseEntity);
+
+        ResponseEntity response =
+                controller.update(loanRest, COMPANY_ACCOUNT_ID, LOANS_ID, request);
+
+        assertNotNull(response);
+        assertEquals(HttpStatus.NO_CONTENT, response.getStatusCode());
+        assertNull(response.getBody());
+
+        verify(loanService, times(1))
+                .update(loanRest, transaction, COMPANY_ACCOUNT_ID, request);
+        verify(apiResponseMapper, times(1))
+                .map(responseObject.getStatus(), responseObject.getData(), responseObject.getErrors());
+    }
+
+    @Test
+    @DisplayName("Tests the update of a Loan when the Loan ID doesnt exist")
+    void UpdateDirectorResourceWhenIdIsNull() throws DataException {
```

Comment:
method name should start with a lowercase letter -updateDirectorResourceWhenIdIsNull

Reply:
Done 👍